### PR TITLE
[fix](move-memtable) check segment num when closing each tablet

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Previously, there is chance that sender failed to send some data while the receiver being unaware of.
This will cause lost data if some segments are skipped.

This PR fixes the problem by including checks in both sender and receiver.
When sender failed to send rpc, LoadStreamStub will mark the involved tablets failed.
Each sender will send segment num for each tablet in CLOSE_LOAD,
and receivers (LoadStream) will sum up and check total segment nums.

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

File: be/src/runtime/load_stream.cpp [hunk 1]
```diff
@@ -307,11 +312,17 @@ Status IndexStream::close(const std::vector<PTabletID>& tablets_to_commit,
...
     // open all need commit tablets
     for (const auto& tablet : tablets_to_commit) {
+        if (_id != tablet.index_id()) {
+            continue;
+        }
         TabletStreamSharedPtr tablet_stream;
         auto it = _tablet_streams_map.find(tablet.tablet_id());
-        if (it == _tablet_streams_map.end() && _id == tablet.index_id()) {
+        if (it == _tablet_streams_map.end()) {
             RETURN_IF_ERROR(
```
Comment (liaoxin01) [inline be/src/runtime/load_stream.cpp:321]: It should be added to the list of failed tablets instead of directly returning a failure.

**Comment by liaoxin01** [comment]
Please add an injection regression case.

File: be/src/runtime/load_stream.cpp (hunk 1 above)
Comment (kaijchen) [reply be/src/runtime/load_stream.cpp:321]: This PR doesn't change the original logic.
Let's fix this in a later PR.

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**